cmd/snap/cmd_run.go

```diff
@@ -1183,8 +1198,6 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 	if needsTracking {
 		opts := &cgroup.TrackingOptions{AllowSessionBus: allowSessionBus}
 		trackingErr = cgroupCreateTransientScopeForTracking(securityTag, opts)
-	} else {
-		trackingErr = cgroupConfirmSystemdServiceTracking(securityTag)
 	}
 	if trackingErr != nil {
```

Comment:
Since there's only one case now, this error handling can be moved under `if needsTracking {`